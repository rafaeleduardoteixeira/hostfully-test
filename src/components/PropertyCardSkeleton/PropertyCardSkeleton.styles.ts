import styled from 'styled-components';
import { media } from '../../themes';

export const Container = styled.div`
  width: 300px;
  height: 330px;
  border: 1px solid #e0e0e0;
  border-radius: 10px;

  @media ${media.tablet} {
    width: 400px;
  }

  @media ${media.mobile} {
    width: 330px;
  }
`;

export const PropertyCardImgSkeleton = styled.div`
  height: 200px;
  width: 100%;
`;

export const PropertyCardDataSkeleton = styled.div`
  display: flex;
  flex-direction: column;
  gap: 5px;
  padding: 15px 20px;
  height: 100px;
`;
