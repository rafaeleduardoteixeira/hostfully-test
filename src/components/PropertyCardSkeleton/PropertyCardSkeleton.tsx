import React from 'react';
import { Container, PropertyCardDataSkeleton, PropertyCardImgSkeleton } from './PropertyCardSkeleton.styles';
import { Skeleton } from '../Skeleton/Skeleton';

export const PropertyCardSkeleton = () => {
  return (
    <Container>
      <PropertyCardImgSkeleton>
        <Skeleton />
      </PropertyCardImgSkeleton>
      <PropertyCardDataSkeleton>
        <Skeleton />
        <Skeleton width="80%" />
        <Skeleton width="40%" />
      </PropertyCardDataSkeleton>
    </Container>
  );
};
