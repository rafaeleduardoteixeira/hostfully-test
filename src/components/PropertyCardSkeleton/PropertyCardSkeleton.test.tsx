import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { PropertyCardSkeleton } from './PropertyCardSkeleton';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('PropertyCardSkeleton', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(<PropertyCardSkeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should render PropertyCardSkeleton', () => {
    render(<PropertyCardSkeleton />);
    const PropertyCardSkeletonElement = screen.getByTestId('PropertyCardSkeleton');
    expect(PropertyCardSkeletonElement).toBeInTheDocument();
  });
});
