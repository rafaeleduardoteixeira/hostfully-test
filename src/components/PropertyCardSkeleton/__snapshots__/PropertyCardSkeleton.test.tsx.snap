// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCardSkeleton Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu gblkLp"
    data-testid="PropertyCardSkeleton"
  >
    <div
      class="sc-gJhJTp hEybUI"
    >
      <div
        class="sc-fQpRED eKGiQ"
      />
    </div>
    <div
      class="sc-elDIKY biXTKv"
    >
      <div
        class="sc-fQpRED eKGiQ"
      />
      <div
        class="sc-fQpRED leCQl"
        width="80%"
      />
      <div
        class="sc-fQpRED gBrIUN"
        width="40%"
      />
    </div>
  </div>
</DocumentFragment>
`;
